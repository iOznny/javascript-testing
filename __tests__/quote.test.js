import Citas from '../testing-jest/js/classes/Citas';

describe('Testing class citas', () => {

    // Init class
    const quotes = new Citas();
    const id = Date.now();

    test('Add new quote', () => {
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Demo',
            telefono: '1234567890',
            fecha: '01-01-2020',
            hora:'01:00',
            sintomas: 'Duerme'
        };

        // Add quote
        quotes.agregarCita(citaObj);

        // Resolve
        expect(quotes).toMatchSnapshot();
    });

    // Update quote
    test('Update quote', () => {
        const updateQuote = {
            id,
            mascota: 'Luke',
            propietario: 'Demo',
            telefono: '1234567890',
            fecha: '01-01-2020',
            hora:'01:00',
            sintomas: 'Duerme'
        };

        quotes.editarCita(updateQuote);

        // Update quote & update snapshot
        expect(quotes).toMatchSnapshot();
    });

    // Delete quote
    test('Delete quote', () => {
        quotes.eliminarCita(id);

        // Update quote & update snapshot
        expect(quotes).toMatchSnapshot();
    });
});