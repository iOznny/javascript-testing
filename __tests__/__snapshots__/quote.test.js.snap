// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing class citas Add new quote 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "01-01-2020",
      "hora": "01:00",
      "id": 1648236260616,
      "mascota": "Hook",
      "propietario": "Demo",
      "sintomas": "Duerme",
      "telefono": "1234567890",
    },
  ],
}
`;

exports[`Testing class citas Delete quote 1`] = `
Citas {
  "citas": Array [],
}
`;

exports[`Testing class citas Update quote 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "01-01-2020",
      "hora": "01:00",
      "id": 1648236260616,
      "mascota": "Luke",
      "propietario": "Demo",
      "sintomas": "Duerme",
      "telefono": "1234567890",
    },
  ],
}
`;
